Source/src/modules/ModuleSceneManager.h

```diff
@@ -52,12 +54,15 @@ namespace Hachiko
         void CreateEmptyScene();
 
         void LoadScene(const char* file_path);
-        void SaveScene(const char* file_path) const;
+        void SaveScene();
+        void SaveScene(const char* path);
 
         void SwitchTo(const char* file_path);
 
     private:
         Scene* main_scene = nullptr;
+        SceneSerializer serializer;
```

Comment:
note that here we are instantiating the serializer, since it's not a pointer

Reply:
should we change it, though??